File: P1S3/src/FiltroEtiquetas.cpp
#include "FiltroEtiquetas.h"
#include <vector>
#include <string>

std::string FiltroEtiquetas::ejecutar(Post &post) {
	
	std::string etiqueta;
	std::vector<std::string> etiquetas;
	std::string texto = post.getTexto();
	int i = 0;

	while(i < texto.size()){
		if(texto[i] == '#'){
                    i++;
			while(texto[i] != ' ' && i < texto.size()){
				etiqueta += texto[i];
				i++;
			}

			if(!etiqueta.empty()){
				etiquetas.push_back(etiqueta);
				etiqueta.clear();
			}
		}
		else
			i++;
	}

	post.setEtiquetas(etiquetas);
        
        return texto;
}

File: P2/FiltroEtiquetas.dart
import 'package:p2/Filtro.dart';
import 'package:p2/Post.dart';

class FiltroEtiquetas extends Filtro {
  ejecutar(Post post) {

    String etiqueta;
    List<String> etiquetas;
    String texto = post.getTexto();
    int i = 0;

    while(i < texto.length){
      if(texto[i] == '#'){
        i++;
        while(texto[i] != ' ' && i < texto.length){
          etiqueta += texto[i];
          i++;
        }

        if(!etiqueta.isEmpty){
          etiquetas.add(etiqueta);
          etiquetas.clear();
        }
    }

    else
      i++;
    }

    post.setEtiquetas(etiquetas);

    return texto;
  }
}